src/Client.ts

```diff
@@ -317,15 +318,20 @@ class Client {
   }
 
   async upload(options: UploadInputType) {
-    const { externalColumnId, file } = options
+    const { file, attribute, resource } = options
     const prepareAssetParams = {
-      externalColumnId: String(externalColumnId)
+      name: file.name,
+      size: file.size,
+      mimeType: file.type,
+      attribute,
+      resource,
+      targetEnvironment: this.targetEnvironment
     }
 
-    const response = await this.makeHttpRequest(prepareExternalAssetRequest, { input: prepareAssetParams })
+    const response = await this.makeHttpRequest(prepareAssetRequest, { input: prepareAssetParams })
 
-    const id = response?.prepareExternalAsset?.id
-    const url = response?.prepareExternalAsset?.data?.upload?.url
+    const id = response?.prepareAsset?.id
+    const url = response?.prepareAsset?.data?.upload?.url
```

Comment:
I think we need to have some error handling

```suggestion
   if (!response?.prepareAsset?.data) {
     // do something and early return
   }

    const id = response.prepareAsset.id
    const url = response.prepareAsset.data.upload.url
```